src/renderer/atlas/AtlasEngine.cpp

```diff
@@ -433,30 +434,30 @@ try
 
     if (options.isOn)
     {
-        const auto point = options.coordCursor;
-        // TODO: options.coordCursor can contain invalid out of bounds coordinates when
-        // the window is being resized and the cursor is on the last line of the viewport.
-        const auto top = clamp(point.y, 0, _p.s->cellCount.y - 1);
-        const auto bottom = top + 1;
         const auto cursorWidth = 1 + (options.fIsDoubleWidth & (options.cursorType != CursorType::VerticalBar));
+        const auto top = options.coordCursor.y;
+        const auto bottom = top + 1;
+        const auto shift = gsl::narrow_cast<u8>(_p.rows[top]->lineRendition != LineRendition::SingleWidth);
+        auto left = options.coordCursor.x - (_p.s->viewportOffset.x >> shift);
+        auto right = left + cursorWidth;
+
+        left <<= shift;
+        right <<= shift;
+
+        _p.cursorRect = {
+            std::max<til::CoordType>(left, 0),
+            std::max<til::CoordType>(top, 0),
+            std::min<til::CoordType>(right, _p.s->viewportCellCount.x),
+            std::min<til::CoordType>(bottom, _p.s->viewportCellCount.y),
+        };
 
-        auto left = std::max(point.x, 0);
-        auto right = std::max(left + cursorWidth, 0);
-
-        if (_p.rows[top]->lineRendition != LineRendition::SingleWidth)
+        if (_p.cursorRect)

```

Comment:
Just like above we need to shift to the right, subtract and and then shift to the left. The only additional difference here is that we now need to check if the `cursorRect` is empty (out of bounds), because the cursor might be on, but simply scrolled off to the side.